### PR TITLE
Plotly Integration

### PR DESCRIPTION
**Pull request by karl-richter**
Based on #322 

**Status quo:**
Currently, NP uses matplotlib as plotting backend. [manikTharaka](https://github.com/manikTharaka) added plotting functions to recreate the matplotlib plots in ploty. Plotly provides a more flexible interface that allows for interactivity. However, as the majority of changes have been performed 16+ months ago, some parts of the implementation are out of date.

**Changes:**
This pr aims to prepare the release of ploty as plotting backend by adding a feature flag for the near term with the option of replacing the matplotlib implementation completely. The following changes are implemented in addition to #322:
- Removed old artifact code
- Added a feature flag to switch between matplotlib and plotly
- Adjusted the plotly plots to be more clean and use the NP colours
- Fixed various bugs in the plotting
- Minor feature improvements

**Open Todos:**
- [x] Check the plotting function documentation (docstrings)
- [x] Correct the weekly seasonality plot to start and end on Sunday
- [x] Align the seasonality curve shape between matplotlib and plotly
- [x] Check which notebooks are still needed (eg. notebooks under example/ can likely be removed)
- [x] Add functionality to set the global forecasting backend via `set_plotting_backend()`
- [x] Centralise the Plotly UI parameters
- [x] Re-styled the plotly plots
- [x] Created a dynamic x-axis range padding function for plotly plots

**Open questions:**
- Should the feature flag be set one during the init of the NP object or provided with every plotting function call?
   Answer: Both should be possible.

### CODE REVIEW

File: neuralprophet/plot_forecast.py [hunk 1]
```diff
@@ -374,13 +374,7 @@ def plot_forecast_component(
...
         if add_x or sum(fcst[comp_name].notna()) == 1:
             artists += ax.plot(fcst_t, y, "bx")
-    # Specify formatting to workaround matplotlib issue #12925
-    locator = AutoDateLocator(interval_multiples=False)
-    formatter = AutoDateFormatter(locator)
-    ax.xaxis.set_major_locator(locator)
-    ax.xaxis.set_major_formatter(formatter)
-    ax.grid(True, which="major", c="gray", ls="-", lw=1, alpha=0.2)
-    ax.set_xlabel("ds")
+
```
Comment (karl-richter) [inline neuralprophet/plot_forecast.py:377]: todo: need to check whether the removal of this functionality is intentional

**Comment by karl-richter** [comment]
**Latest adaptations:**
- [ ] Moved the global forecasting backend flag to `set_plotting_backend()`
- [ ] Centralised the plotly UI adaptions
- [ ] Re-styled the plotly plots
- [ ] Created a dynamic range padding function for plotly

**Comment by karl-richter** [comment]
@ourownstory the branch would be ready for a first review from my side.
I am still performing minor bugfixes but feel free to give feedback on the implementation approach already.

**Comment by karl-richter** [comment]
@ourownstory the plotly pytests are now separated and the PR should be ready for review! 🙂

**Comment by ourownstory** [comment]
Thank you!

**Pull request merged**